src/Razor/src/Microsoft.VisualStudio.RazorExtension/Snippets/SnippetService.cs

```diff
@@ -0,0 +1,215 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.Editor.Razor;
+using Microsoft.VisualStudio.Editor.Razor.Snippets;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.RazorExtension.Snippets;
+
+internal class SnippetService
+{
+    private readonly JoinableTaskFactory _joinableTaskFactory;
+    private readonly IAsyncServiceProvider _serviceProvider;
+    private readonly SnippetCache _snippetCache;
+    private IVsExpansionManager? _vsExpansionManager;
+
+    private readonly object _cacheGuard = new();
+
+    private static readonly Guid s_CSharpLanguageId = new("694dd9b6-b865-4c5b-ad85-86356e9c88dc");
+    private static readonly Guid s_HtmlLanguageId = new("9bbfd173-9770-47dc-b191-651b7ff493cd");
```

Comment:
Is there a `Guids` class somewhere that these could be moved to?

Reply:
I didn't see one. Could add to `RazorConstants` maybe? But right now they're not defined or used anywhere else